File: Problems/No.75_Sort_Colors/sort_colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        
        // parameters
        int cur = 0;
        int left = 0;
        int right = nums.size() - 1;

        // start computing (use Dutch National Flag algorithm)
        while (cur <= right) 
            switch(nums[cur]) {

                case 0:
                    std::swap(nums[cur], nums[left]);
                    left++;
                    cur++;
                    break;

                case 1:
                    cur++;
                    break;

                case 2:
                    std::swap(nums[cur], nums[right]);
                    right--;
                    break;
                    
                default:
                    break;

            }

    }
};

File: Problems/No.75_Sort_Colors/sort_colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """

        # parameters
        cur, left, right = 0, 0, len(nums)-1

        # start computing (use Dutch National Flag algorithm)
        while (cur <= right):
            if (nums[cur] == 0):
                nums[cur], nums[left] = nums[left], nums[cur]
                left += 1
                cur += 1
            elif (nums[cur] == 1):
                cur += 1
            else:
                nums[cur], nums[right] = nums[right], nums[cur]
                right -= 1
